File: 151051/Makovetskiy/task330+340+350/dc-rest/src/main/java/com/rmakovetskij/dc_rest/mapper/IssueMapper.java
package com.rmakovetskij.dc_rest.mapper;

import com.rmakovetskij.dc_rest.model.Issue;
import com.rmakovetskij.dc_rest.model.dto.requests.IssueRequestTo;
import com.rmakovetskij.dc_rest.model.dto.responses.IssueResponseTo;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface IssueMapper {

    @Mapping(source = "editor.id", target = "editorId")
    IssueResponseTo toResponse(Issue issue);

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "created", ignore = true)
    @Mapping(target = "modified", ignore = true)
    Issue toEntity(IssueRequestTo issueRequestDto);
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/mapper/IssueMapper.kt
package bsuir.dc.publisher.mapper

import bsuir.dc.publisher.dto.from.IssueRequestTo
import bsuir.dc.publisher.dto.to.IssueResponseTo
import bsuir.dc.publisher.entity.Issue
import bsuir.dc.publisher.entity.Label
import bsuir.dc.publisher.entity.Writer

fun IssueRequestTo.toEntity(writer: Writer, labels: MutableSet<Label>) = Issue(
    id = this.id,
    writer = writer,
    title = this.title,
    content = this.content,
    labels = labels,
)

fun Issue.toResponse() = IssueResponseTo(
    id = this.id,
    writerId = this.writer.id,
    title = this.title,
    content = this.content,
    created = this.created,
    modified = this.modified
)
